The OP_DROP_TABLE_V2 event is not triggered when an error is reported for dropping a materialized view.

### Steps to reproduce the behavior (Required)
1. create hive table in 172.17.80.138
```
CREATE TABLE test.sales2 (
    id INT,
    product STRING,
    amount FLOAT
)
PARTITIONED BY (sale_date STRING) 
ROW FORMAT SERDE
'org.apache.hadoop.hive.ql.io.orc.OrcSerde'
STORED AS INPUTFORMAT
'org.apache.hadoop.hive.ql.io.orc.OrcInputFormat'
OUTPUTFORMAT
'org.apache.hadoop.hive.ql.io.orc.OrcOutputFormat'
TBLPROPERTIES (
'orc.compress'='LZ4',
'transient_lastDdlTime'='1669911897');
```
2. create hive catalog
```
CREATE EXTERNAL CATALOG prod_hive
PROPERTIES
(
    "type" = "hive",
    "hive.metastore.type" = "hive",
    "hive.metastore.uris" = "thrift://	172.17.80.138:7004"
);
```
3. create mv
```
CREATE MATERIALIZED VIEW test.dws_ofo_mv4
REFRESH ASYNC START('2024-12-01 09:00:00') EVERY (interval 1 day)
PARTITION BY str2date(sale_date, '%Y-%m-%d')
AS
SELECT sale_date, id, product, amount
FROM prod_hive.test.sales2;
```

4. drop hive catalog
```
 drop CATALOG prod_hive;
```
5 create another hive catalog
```
CREATE EXTERNAL CATALOG prod_hive
PROPERTIES
(
    "type" = "hive",
    "hive.metastore.type" = "hive",
    "hive.metastore.uris" = "thrift://172.17.48.9:7004"
);
```
6 
Metastroe does not yet have test.sales2 in the node 172.17.48.9
```
drop MATERIALIZED VIEW test.dws_ofo_mv4;
```

### Real behavior (Required)
<img width="959" alt="Clipboard_Screenshot_1734279767" src="https://github.com/user-attachments/assets/5c65e7a6-a4df-4f03-88e8-7ff0bbca52ec" />


```
2024-12-13 11:21:39.548+08:00 ERROR (starrocks-mysql-nio-pool-1|17390) [HiveMetadata.getTable():196] Failed to get hive table [prod_hive.test.sales2]
com.starrocks.connector.exception.StarRocksConnectorException: Failed to get table [test.sales2], msg: NoSuchObjectException: hive.test.sales2 table not found
        at com.starrocks.connector.hive.HiveMetaClient.callRPC(HiveMetaClient.java:166) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.hive.HiveMetaClient.callRPC(HiveMetaClient.java:151) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.hive.HiveMetaClient.getTable(HiveMetaClient.java:260) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.hive.HiveMetastore.getTable(HiveMetastore.java:117) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.hive.CachingHiveMetastore.loadTable(CachingHiveMetastore.java:269) ~[starrocks-fe.jar:?]
        at com.google.common.cache.CacheLoader$FunctionToCacheLoader.load(CacheLoader.java:169) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.CacheLoader$1.load(CacheLoader.java:192) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3570) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2312) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2189) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2079) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache.get(LocalCache.java:4011) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4034) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5010) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:5017) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.connector.metastore.CachingMetastore.get(CachingMetastore.java:88) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.hive.CachingHiveMetastore.getTable(CachingHiveMetastore.java:261) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.hive.CachingHiveMetastore.loadTable(CachingHiveMetastore.java:269) ~[starrocks-fe.jar:?]
        at com.google.common.cache.CacheLoader$FunctionToCacheLoader.load(CacheLoader.java:169) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.CacheLoader$1.load(CacheLoader.java:192) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3570) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2312) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2189) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2079) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache.get(LocalCache.java:4011) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4034) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5010) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:5017) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.connector.metastore.CachingMetastore.get(CachingMetastore.java:88) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.hive.CachingHiveMetastore.getTable(CachingHiveMetastore.java:261) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.hive.HiveMetastoreOperations.getTable(HiveMetastoreOperations.java:238) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.hive.HiveMetadata.getTable(HiveMetadata.java:194) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.CatalogConnectorMetadata.getTable(CatalogConnectorMetadata.java:135) ~[starrocks-fe.jar:?]
        at com.starrocks.server.MetadataMgr.lambda$getTable$5(MetadataMgr.java:497) ~[starrocks-fe.jar:?]
        at java.util.Optional.map(Optional.java:265) ~[?:?]
        at com.starrocks.server.MetadataMgr.getTable(MetadataMgr.java:497) ~[starrocks-fe.jar:?]
        at com.starrocks.server.MetadataMgr.getTable(MetadataMgr.java:526) ~[starrocks-fe.jar:?]
        at com.starrocks.server.MetadataMgr.getTableWithIdentifier(MetadataMgr.java:531) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.rule.transformation.materialization.MvUtils.getTableWithIdentifier(MvUtils.java:1413) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.MaterializedView.onDrop(MaterializedView.java:899) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.Database.unprotectDropTable(Database.java:380) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.Database.dropTable(Database.java:329) ~[starrocks-fe.jar:?]
        at com.starrocks.server.LocalMetastore.dropMaterializedView(LocalMetastore.java:3399) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.DDLStmtExecutor$StmtExecutorVisitor.lambda$visitDropMaterializedViewStatement$13(DDLStmtExecutor.java:356) ~[starrocks-fe.jar:?]
        at com.starrocks.common.ErrorReport.wrapWithRuntimeException(ErrorReport.java:108) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.DDLStmtExecutor$StmtExecutorVisitor.visitDropMaterializedViewStatement(DDLStmtExecutor.java:355) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.DDLStmtExecutor$StmtExecutorVisitor.visitDropMaterializedViewStatement(DDLStmtExecutor.java:182) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.DropMaterializedViewStmt.accept(DropMaterializedViewStmt.java:63) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.AstVisitor.visit(AstVisitor.java:71) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.DDLStmtExecutor.execute(DDLStmtExecutor.java:168) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.handleDdlStmt(StmtExecutor.java:1702) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:690) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:353) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:548) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:882) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:69) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
        at java.lang.Thread.run(Thread.java:829) ~[?:?]
Caused by: java.lang.reflect.InvocationTargetException
        at jdk.internal.reflect.GeneratedMethodAccessor11.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at com.starrocks.connector.hive.HiveMetaClient.callRPC(HiveMetaClient.java:162) ~[starrocks-fe.jar:?]
        ... 58 more
Caused by: org.apache.hadoop.hive.metastore.api.NoSuchObjectException: hive.test.sales2 table not found
        at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$get_table_result$get_table_resultStandardScheme.read(ThriftHiveMetastore.java) ~[hive-apache-3.1.2-22.jar:?]
        at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$get_table_result$get_table_resultStandardScheme.read(ThriftHiveMetastore.java) ~[hive-apache-3.1.2-22.jar:?]
        at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$get_table_result.read(ThriftHiveMetastore.java) ~[hive-apache-3.1.2-22.jar:?]
        at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:93) ~[libthrift-0.20.0.jar:0.20.0]
        at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$Client.recv_get_table(ThriftHiveMetastore.java:2042) ~[hive-apache-3.1.2-22.jar:?]
        at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$Client.get_table(ThriftHiveMetastore.java:2028) ~[hive-apache-3.1.2-22.jar:?]
        at org.apache.hadoop.hive.metastore.HiveMetaStoreClient.getTable(HiveMetaStoreClient.java:638) ~[starrocks-fe.jar:?]
        at org.apache.hadoop.hive.metastore.HiveMetaStoreClient.getTable(HiveMetaStoreClient.java:630) ~[starrocks-fe.jar:?]
        at jdk.internal.reflect.GeneratedMethodAccessor14.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.apache.hadoop.hive.metastore.RetryingMetaStoreClient.invoke(RetryingMetaStoreClient.java:208) ~[hive-apache-3.1.2-22.jar:?]
        at com.sun.proxy.$Proxy35.getTable(Unknown Source) ~[?:?]
        at jdk.internal.reflect.GeneratedMethodAccessor11.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at com.starrocks.connector.hive.HiveMetaClient.callRPC(HiveMetaClient.java:162) ~[starrocks-fe.jar:?]
        ... 58 more
2024-12-13 11:21:39.549+08:00 INFO (starrocks-mysql-nio-pool-1|17390) [StmtExecutor.execute():742] execute Exception, sql: drop  MATERIALIZED VIEW test.dws_ofo_mv4, error: 
2024-12-13 11:21:39.549+08:00 INFO (starrocks-mysql-nio-pool-1|17390) [MetadataMgr.removeQueryMetadata():200] Succeed to deregister query level connector metadata on query id: Optional[5def3c23-b901-11ef-a681-5254000e4d4e]
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

--- a/fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java
+++ b/fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java
@@ -957,7 +957,22 @@ public class MaterializedView extends OlapTable implements GsonPreProcessable, G
         // 2. Remove from base tables
         List<BaseTableInfo> baseTableInfos = getBaseTableInfos();
         for (BaseTableInfo baseTableInfo : ListUtils.emptyIfNull(baseTableInfos)) {
-            Optional<Table> baseTableOpt = MvUtils.getTableWithIdentifier(baseTableInfo);
+            Optional<Table> baseTableOpt;
+            try {
+                baseTableOpt = MvUtils.getTableWithIdentifier(baseTableInfo);
+            } catch (Exception e) {
+                if (!(baseTableInfo.isInternalCatalog())) {
+                    GlobalStateMgr.getCurrentState().getConnectorTblMetaInfoMgr().
+                            removeConnectorTableInfo(baseTableInfo.getCatalogName(),
+                                    baseTableInfo.getDbName(),
+                                    baseTableInfo.getTableIdentifier(),
+                                    ConnectorTableInfo.builder().setRelatedMaterializedViews(
+                                            Sets.newHashSet(mvId)).build());
+                }
+                LOG.error("Failed to get base table: {}", baseTableInfo, e);
+                continue;
+            }
+
             if (baseTableOpt.isPresent()) {
                 Table baseTable = baseTableOpt.get();
                 baseTable.removeRelatedMaterializedView(mvId);

--- a/fe/fe-core/src/main/java/com/starrocks/qe/DDLStmtExecutor.java
+++ b/fe/fe-core/src/main/java/com/starrocks/qe/DDLStmtExecutor.java
@@ -178,7 +178,8 @@ public class DDLStmtExecutor {
             } else if (re.getCause() instanceof IOException) {
                 throw (IOException) re.getCause();
             } else if (re.getCause() != null) {
-                throw new DdlException(re.getCause().getMessage(), re);
+                throw new DdlException(re.getCause().getMessage() != null ?
+                        re.getCause().getMessage() : re.getMessage(), re);
             } else {
                 throw re;
             }

